core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
 
   newGauge("IdlePercent",
```

Comment:
Thanks, got it !
![image](https://cloud.githubusercontent.com/assets/6661532/26005185/33e1f2e8-3730-11e7-8350-306f5ae7d3cf.png)

It's still a per-network-processor-thread metric. That's why the coding was intentional,
though I did not notice the new MBeanName surfacing in a different place.

I prefer the tagging, but I still could live without it here and maintain compatibility. 
Please advise, @ijuma @junrao